triton/tail.go

```diff
@@ -0,0 +1,236 @@
+package triton
+
+import (
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	"github.com/tinylib/msgp/msgp"
+	"io"
+	"sync"
+	"time"
+)
+
+type TailAt struct {
+	stream            string
+	streamInited      bool
+	bucket            string
+	client            string
+	at                time.Time
+	closed            bool
+	records           chan Record
+	errors            chan error
+	region            string
+	kinesisService    KinesisService
+	s3Service         S3Service
+	pollInterval      time.Duration
+	emptyPollInterval time.Duration
+	sync.Mutex
+}
+
+// NewTailAtParams are the parameters for invoking NewTailAt
+type NewTailAtParams struct {
+	S3Service         S3Service
+	KinesisService    KinesisService
+	StreamName        string
+	Bucket            string
+	Client            string
+	At                time.Time
+	PollInterval      time.Duration
+	EmptyPollInterval time.Duration
+}
+
+const (
+	// The amount of time to poll for new kinesis records if we encounter an empty stream
+	DefaultEmptyPollInterval = 10 * time.Second
+	DefaultPollInterval      = 1 * time.Millisecond
+)
+
+// NewTail returns a new tailing stream starting at "at"
```

Comment:
This might be more instructive as:

"NewTail returns a new stream starting at the specified time" or something like that? "at" seems ambiguous.